superset-frontend/packages/superset-ui-chart-controls/src/types.ts

```diff
@@ -65,6 +65,11 @@ export interface DatasourceMeta {
   granularity_sqla?: string;
   datasource_name: string | null;
   description: string | null;
+  sl_dataset?: {
+    sl_table?: any;
```

Comment:
can we cast these to values we actually want them to be

Reply:
Do you know what the types for these will be? I was going to figure that out once the backend came through and change it then.

Reply:
@eschutho shouldn't the model look like this since sl_dataset it's own entity
```
sl_datasource?: {
  sl_dataset: any;
  sl_table?: any;
  query?: any;
  saved_query?: any;
};
```

Reply:
I think these values would actually be string options of `type`

Reply:
So it will look like this?

```
sl_datasource?: {
  sl_dataset: string;
  sl_table?: string;
  query?: string;
  saved_query?: string;
};
```

Reply:
I think you can delete all that.. it's all on the type property here under `DatasourceType`
```
export interface DatasourceMeta {
  id: number;
  type: DatasourceType;
  ```

Reply:
If I delete this, how would I access the data in cases like this one? I think `DatasourceType` only defines the types
https://github.com/apache/superset/blob/9985ec71158a6d7088f1382b24bfa1454d4891ea/superset-frontend/src/explore/components/DatasourcePanel/index.tsx#L253

Reply:
On a query, for example, the data structure will look like `datasource.query` but there won't be a query on a table, so we should define them separately.

Reply:
Is the schema for these types anywhere so I can describe them properly?

Reply:
maybe we can just start with Query: superset-frontend/src/SqlLab/types.ts

Reply:
looks like there's an interface for savedquery here: superset-frontend/src/views/CRUD/welcome/SavedQueries.tsx but don't be confused.. it's named query :)

Reply:
bring this into the component:
```
type ExploreDatasource =
    Dataset |
    Query 
```
Rename `DatasourceMeta` to Dataset